### PR TITLE
enable validating webhook for MCO

### PR DESCRIPTION
**Pull request by morvencao**
This PR:
- enable validating webhook for MCO
- remove unused manifest under `config` directory
- upgrade the service serving cert ca injection annotation from `alpha` to `beta`

### CODE REVIEW

File: operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go [hunk 1]
```diff
@@ -4,24 +4,210 @@
...
+// validateMultiClusterObservabilityName validates the name of the MultiClusterObservability CR.
+// Validating the length of a string field can be done declaratively by the validation schema.
+// But the `ObjectMeta.Name` field is defined in a shared package under the apimachinery repo,
+// so we can't declaratively validate it using the validation schema.
+func (mco *MultiClusterObservability) validateMultiClusterObservabilityName() *field.Error {
+	if len(mco.ObjectMeta.Name) > validationutils.DNS1035LabelMaxLength-43 {
+		// The MCO CR name length is 63 character like all Kubernetes objects (which must fit in a DNS subdomain).
+		// The observatorium controller appends at most a 43-character suffix to the MCO CR (`-thanos-receive-controller-6b446c5576-hj6xl`)
+		// when creating thanos-receive-controller pod. The thanos-receive-controller pod length limit is 63 characters. Therefore MCO CR
+		// names must have length <= 63-43=20. If we don't validate this here, then pod creation will fail later.
```
Comment (clyang82) [inline operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go:92]: Right now, the MCO CR Name is not part of pod name anymore. all the pods' name starts from `observability` which is hardcoded. so do not need to have `20` limitation.

File: operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go [hunk 2]
```diff
@@ -4,24 +4,210 @@
...
+		}
+		if mcoOldConfig.CompactStorageSize != mcoNewConfig.CompactStorageSize {
+			errs = append(errs, field.Forbidden(storageConfigFieldPath.Child("compactStorageSize"), storageForbiddenResize))
+		}
+		if mcoOldConfig.ReceiveStorageSize != mcoNewConfig.ReceiveStorageSize {
+			errs = append(errs, field.Forbidden(storageConfigFieldPath.Child("receiveStorageSize"), storageForbiddenResize))
+		}
+		if mcoOldConfig.StoreStorageSize != mcoNewConfig.StoreStorageSize {
+			errs = append(errs, field.Forbidden(storageConfigFieldPath.Child("storeStorageSize"), storageForbiddenResize))
+		}
```
Comment (clyang82) [inline operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go:150]: Need have `ruleStorageSize` here.

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: operators/multiclusterobservability/bundle/manifests/multicluster-observability-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -51,7 +51,6 @@ metadata:
     capabilities: Basic Install
     operators.operatorframework.io/builder: operator-sdk-v1.4.2
     operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
-    operators.operatorframework.io/internal-objects: '["observatoria.core.observatorium.io","observabilityaddons.observability.open-cluster-management.io"]'
```
[the remark this answers is not in the document]
Comment (morvencao) [reply operators/multiclusterobservability/bundle/manifests/multicluster-observability-operator.clusterserviceversion.yaml:54]: this was added manually?

File: operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go (hunk 1 above)
Comment (morvencao) [reply operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go]: removed name length validation.

File: operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go (hunk 2 above)
Comment (morvencao) [reply operators/multiclusterobservability/api/v1beta2/multiclusterobservability_webhook.go:142]: done.

File: operators/multiclusterobservability/bundle/manifests/multicluster-observability-operator.clusterserviceversion.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (morvencao) [reply operators/multiclusterobservability/bundle/manifests/multicluster-observability-operator.clusterserviceversion.yaml]: somehow the `make bundle` omit this, I added it back manually.

[the remark this answers is not in the document]
Comment (clyang82) [reply operators/multiclusterobservability/bundle/manifests/multicluster-observability-operator.clusterserviceversion.yaml]: Yes. it was added manually.

**Comment by morvencao** [comment]
ref: https://github.com/open-cluster-management/backlog/issues/14103

**Pull request merged**